### ISSUE TITLE
Matrix swizzles not supported in GLSL

### ISSUE THREAD

**Issue by BearishSun** [LukasBanana/XShaderCompiler#52]
When using 'glslang' to compile GLSL -> SPIR-V (which I'm using for Vulkan) it complains when using matrix swizzles:
```
mat3 tangentToWorld;
vec3 normal = vec3(tangentToWorld._m02_m12_m22);
```

Error:
> ERROR: 0:90: '_m02_m12_m22' : does not apply to this type: temp highp 3X3 matrix of float

I'm not getting a similar error when I use the same code in OpenGL. But looking at the specification of GLSL 4.5 (haven't checked earlier versions), there indeed doesn't seem to be a mention of matrix swizzles.

This isn't a major issue for me, but it should get worked around eventually.

**Comment by LukasBanana** [comment]
Thanks for pointing this out. I'll keep an eye on the issue.

**Comment by LukasBanana** [comment]
Should be fixed with [018f741](https://github.com/LukasBanana/XShaderCompiler/commit/018f74103a41bed193e35e44739aa5c75eb4f331).

Matrix subscripts of scalars (e.g. `m._11`) are converted to array accesses, and subscripts of vectors (e.g. `m._11_22`) are converted to function calls of the respective wrapper function.
Here is an example of such a wrapper function:
```hlsl
vec3 xst_subscript_00_31_22(mat4 m) {
    return vec3(m[0][0], m[3][1], m[2][2]);
}
```

I also added a test for matrix subscripts: [MatrixSubscriptsTest1.hlsl](https://github.com/LukasBanana/XShaderCompiler/blob/master/test/MatrixSubscriptsTest1.hlsl).
